File: Java/sga/model/Profesor.java
package sga.model;

public class Profesor extends Persona {
    private String especialidad;
    private String materiaAsignada;

    public Profesor(String cedula, String nombre, String correo, String especialidad, String materiaAsignada) {
        super(cedula, nombre, correo);
        this.especialidad = especialidad;
        this.materiaAsignada = materiaAsignada;
    }

    public String getEspecialidad() {
        return especialidad;
    }

    public String getMateriaAsignada() {
        return materiaAsignada;
    }

    @Override
    public String mostrarInformacion() {
        return String.format("[Profesor] ID: %s | Nombre: %s | Correo: %s | Esp: %s | Materia: %s",
                cedula, nombre, correo, especialidad, materiaAsignada);
    }
}

File: C++/Profesor.cpp
#include "Profesor.h"
#include <sstream>

Profesor::Profesor(const std::string& cedula, const std::string& nombre,
                   const std::string& correo, const std::string& especialidad,
                   const std::string& materiaAsignada)
    : Persona(cedula, nombre, correo), especialidad(especialidad), materiaAsignada(materiaAsignada) {}

std::string Profesor::getEspecialidad() const {
    return especialidad;
}

std::string Profesor::getMateriaAsignada() const {
    return materiaAsignada;
}

std::string Profesor::mostrarInformacion() const {
    std::ostringstream oss;
    oss << "[Profesor] ID: " << cedula
        << " | Nombre: " << nombre
        << " | Correo: " << correo
        << " | Esp: " << especialidad
        << " | Materia: " << materiaAsignada;
    return oss.str();
}